### PR TITLE
added documentation for getting started with wsl2 on windows

### PR DESCRIPTION
**Pull request by danielgrzenda**
### Overview

Added documentation for getting started with wsl2 on windows. Added directions for: 

1. running `jupyter lab` with VTK and headless Xserver
2. running `jupyter lab` with `pythreejs`

### CODE REVIEW

File: doc/getting-started/installation.rst [hunk 1]
```diff
@@ -247,10 +246,101 @@ Reconnect to the server with port-forwarding, and start Jupyter:
 
     ssh -i "your-ssh-key" your-user-name@your-server-ip -L 8888:localhost:8888
     conda activate vtk_env
-    jupyter notebook --NotebookApp.token='' --no-browser --port=8888
+    jupyter lab --NotebookApp.token='' --no-browser --port=8888
+
+Visit ``localhost:8888`` in the web browser.
+
+Running on WSL
+~~~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (tkoyama010) [inline doc/getting-started/installation.rst:254]:
```suggestion
Running on WSL
~~~~~~~~~~~~~~
```

File: doc/getting-started/installation.rst [hunk 2]
```diff
@@ -247,10 +246,101 @@ Reconnect to the server with port-forwarding, and start Jupyter:
...
+    conda install nodejs  # required when importing pyvista in Jupyter
+    pip install jupyter pyvista ipyvtklink
+
+    # To avoid "ModuleNotFoundError: No module named 'vtkOpenGLKitPython' " when importing vtk
+    # https://stackoverflow.com/q/32389599
+    # https://askubuntu.com/q/629692
+    sudo apt update && sudo apt install python-qt4 libgl1-mesa-glx
+
+VTK Link to Jupyter
+^^^^^^^^^^^^^^^^^^^^^^
```
Comment (tkoyama010) [inline doc/getting-started/installation.rst:278]:
```suggestion
VTK Link to Jupyter
^^^^^^^^^^^^^^^^^^^
```

File: doc/getting-started/installation.rst [hunk 3]
```diff
@@ -247,10 +246,101 @@ Reconnect to the server with port-forwarding, and start Jupyter:
...
+    actor = pl.add_mesh(pyvista.Cube())
+    pl.subplot(0, 1)
+    actor = pl.add_mesh(pyvista.Sphere())
+    pl.set_background('orange', all_renderers=False)
+    pl.show()
+
+You're visualizations should now be showing directly in the jupyter frontend.
+
+PyThreeJS Rendering in Jupyter
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
Comment (tkoyama010) [inline doc/getting-started/installation.rst:318]:
```suggestion
PyThreeJS Rendering in Jupyter
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

File: doc/getting-started/installation.rst [hunk 4]
```diff
@@ -247,10 +246,101 @@ Reconnect to the server with port-forwarding, and start Jupyter:
...
+    import pyvista
+    pyvista.global_theme.jupyter_backend='pythreejs'
+    pl = pyvista.Plotter(shape=(1, 2))
+    actor = pl.add_mesh(pyvista.Cube())
+    pl.subplot(0, 1)
+    actor = pl.add_mesh(pyvista.Sphere())
+    pl.set_background('orange', all_renderers=False)
+    pl.show()
+
+You're visualizations should now be showing directly in the jupyter frontend.
```
Comment (tkoyama010) [inline doc/getting-started/installation.rst:343]:
```suggestion
Your visualizations should now be showing directly in the Jupyter frontend.
```

File: doc/getting-started/installation.rst [hunk 5]
```diff
@@ -247,10 +246,101 @@ Reconnect to the server with port-forwarding, and start Jupyter:
...
-    jupyter notebook --NotebookApp.token='' --no-browser --port=8888
+    jupyter lab --NotebookApp.token='' --no-browser --port=8888
+
+Visit ``localhost:8888`` in the web browser.
+
+Running on WSL
+~~~~~~~~~~~~~~~~~~~~~~~~~
+Similar to the remote servers example above, the windows subsystem for linux does
+not provide and x-server for visualization. Instead the fastest way to get up and
+running on WSL is through ``jupyter lab``.
```
Comment (tkoyama010) [inline doc/getting-started/installation.rst:257]:
```suggestion
Similar to the example of the remote server above, the windows subsystem for Linux does
not provide an x-server for visualization. Instead, the fastest way to get up and
running on WSL is through ``jupyter lab``.
```

File: doc/getting-started/installation.rst [hunk 6]
```diff
@@ -247,10 +246,101 @@ Reconnect to the server with port-forwarding, and start Jupyter:
...
+
+First make sure you have installed the correct environment through Miniconda and
+related packages:
```
Comment (tkoyama010) [inline doc/getting-started/installation.rst:260]:
```suggestion
First, make sure you have installed the correct environment through Miniconda and
related packages:
```

File: doc/getting-started/installation.rst [hunk 7]
```diff
@@ -247,10 +246,101 @@ Reconnect to the server with port-forwarding, and start Jupyter:
...
+Start Jupyter:
+
+.. code-block:: bash
+
+    jupyter lab --NotebookApp.token='' --no-browser --port=8888
+
+Visit ``localhost:8888`` in the web browser.
+
+Finally add this example code, and your interactive visualizations
+should be displayed in ``jupyter lab``.
```
Comment (tkoyama010) [inline doc/getting-started/installation.rst:303]:
```suggestion
Finally add this example code and your interactive visualizations
should be displayed in ``jupyter lab``.
```

File: doc/getting-started/installation.rst [hunk 8]
```diff
@@ -247,10 +246,101 @@ Reconnect to the server with port-forwarding, and start Jupyter:
...
+
+    import pyvista
+    pl = pyvista.Plotter(shape=(1, 2))
+    actor = pl.add_mesh(pyvista.Cube())
+    pl.subplot(0, 1)
+    actor = pl.add_mesh(pyvista.Sphere())
+    pl.set_background('orange', all_renderers=False)
+    pl.show()
+
+You're visualizations should now be showing directly in the jupyter frontend.
```
Comment (tkoyama010) [inline doc/getting-started/installation.rst:315]:
```suggestion
Your visualizations should now be showing directly in the Jupyter frontend.
```

File: doc/getting-started/installation.rst [hunk 9]
```diff
@@ -247,10 +246,101 @@ Reconnect to the server with port-forwarding, and start Jupyter:
...
+
+PyThreeJS Rendering in Jupyter
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+The second option is to change the ``PyVista`` backend to use
+``pythreejs``.
+
+To do this, first launch the jupyter server:
```
Comment (tkoyama010) [inline doc/getting-started/installation.rst:322]:
```suggestion
To do this, first launch the Jupyter server:
```

**Review by tkoyama010 — COMMENTED** [review]
@danielgrzenda Thank you for adding this. Document for wsl2 is a very important contribution. I have some fixes. If you approve it please commit suggestion.

**Review by tkoyama010 — APPROVED** [review]
LGTM

File: doc/getting-started/installation.rst [hunk 10]
```diff
@@ -247,10 +246,101 @@ Reconnect to the server with port-forwarding, and start Jupyter:
...
-    jupyter notebook --NotebookApp.token='' --no-browser --port=8888
+    jupyter lab --NotebookApp.token='' --no-browser --port=8888
 
 Visit ``localhost:8888`` in the web browser.
 
+Running on WSL
+~~~~~~~~~~~~~~
+Similar to the example of the remote server above, the windows subsystem for Linux does
+not provide an x-server for visualization. Instead, the fastest way to get up and
+running on WSL is through ``jupyter lab``.
```
Comment (akaszynski) [inline doc/getting-started/installation.rst:257]:
```suggestion
running on WSL is through `JupyterLab <https://jupyter.org/>`_.
```

File: doc/getting-started/installation.rst [hunk 11]
```diff
@@ -247,10 +246,101 @@ Reconnect to the server with port-forwarding, and start Jupyter:
...
+    pip install jupyter pyvista ipyvtklink
+
+    # To avoid "ModuleNotFoundError: No module named 'vtkOpenGLKitPython' " when importing vtk
+    # https://stackoverflow.com/q/32389599
+    # https://askubuntu.com/q/629692
+    sudo apt update && sudo apt install python-qt4 libgl1-mesa-glx
+
+VTK Link to Jupyter
+^^^^^^^^^^^^^^^^^^^
+There are two ways to get vtk rendering 3D objects in ``jupyter lab``. First you
```
Comment (akaszynski) [inline doc/getting-started/installation.rst:279]: It's up to debate, but I'm following the formatting used in their installation docs at https://jupyterlab.readthedocs.io/en/stable/getting_started/installation.html

```suggestion
There are two ways to get vtk rendering 3D objects in JupyterLab. First you
```

File: doc/getting-started/installation.rst [hunk 12]
```diff
@@ -247,10 +246,101 @@ Reconnect to the server with port-forwarding, and start Jupyter:
...
+Start Jupyter:
+
+.. code-block:: bash
+
+    jupyter lab --NotebookApp.token='' --no-browser --port=8888
+
+Visit ``localhost:8888`` in the web browser.
+
+Finally add this example code and your interactive visualizations
+should be displayed in ``jupyter lab``.
```
Comment (akaszynski) [inline doc/getting-started/installation.rst:303]:
```suggestion
should be displayed in JupyterLab.
```

File: doc/getting-started/installation.rst [hunk 13]
```diff
@@ -247,10 +246,101 @@ Reconnect to the server with port-forwarding, and start Jupyter:
...
+
+To do this, first launch the Jupyter server:
+
+.. code-block:: bash
+
+    jupyter lab --NotebookApp.token='' --no-browser --port=8888
+
+Visit ``localhost:8888`` in the web browser.
+
+Finally change the PyVista backend to a web visualization library: `pythreejs`.
```
Comment (akaszynski) [inline doc/getting-started/installation.rst:330]:
```suggestion
Finally change the PyVista backend to a web visualization library: ``pythreejs``.
```

**Comment by akaszynski** [comment]
Thanks @danielgrzenda!

**Pull request merged**